spec/support/examples/cache.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+shared_examples 'cacheable response' do |expects_vary: true|
+  it 'does not set cookies' do
+    expect(response.cookies).to be_empty
+    expect(response.headers['Set-Cookies']).to be_nil
+  end
+
+  it 'does not set sessions' do
+    expect(session).to be_empty
+  end
+
+  if expects_vary
+    it 'returns Vary header' do
+      expect(response.headers['Vary']).to include 'Accept, Accept-Language, Cookie'
```

Comment:
I would prefer if the value was not hardcoded but came from `expects_vary`.

Reply:
As in ... the cases which DO need a `Vary` value here should explicitly pass in a string with the values they expect to find? (or false when they dont?)

Reply:
Sorry, I realize that I haven't replied to this. Yes, that's what I meant, I suggest `expects_vary` to be either false or a string, and in the latter case have it be used for comparison. Even if that does add a little duplication, I think it's better if it's explicit here.

Reply:
Rebased, and pushed another commit which flips the default to false and requires a string be passed in (which gets checked if present).